add timeline prototype

ChangjunZhoJoe opened the pull request.

ChangjunZhoJoe committed:
add timeline prototype

ChangjunZhoJoe committed:
minor refactor

ParsaYadollahi commented on dsc-website-client/src/App.js:
```diff
@@ -21,6 +21,7 @@ function App() {
           <Route exact path="/contact" component={Contact} />
           <Route exact path="/faq" component={Faq} />
           <Route exact path="/pastevents" component={PastEvents} />
+          <Route exact path="/timeline" component={CustomizedTimeline} />
```
Could you change this name to /pastEvents or something along those lines?

ChangjunZhoJoe replied:
sure!

ParsaYadollahi commented on dsc-website-client/src/pages/CustomizedTimeline.js:
```diff
@@ -0,0 +1,95 @@
+  const classes = useStyles();
+
+  return (
+    <Timeline align="alternate">
```
Do you think you could make this into a looping function? As in have all the elements in a list and iterate through them. Would make the code cleaner and have less lines.
For example:
![Screen Shot 2020-10-24 at 2 38 28 PM](https://user-images.githubusercontent.com/43769966/97091010-96c46400-1606-11eb-9c2b-89bd8d6f291c.png)

If this doesn't work or takes too much time, don't worry about it well merge it and come back to it later!

ChangjunZhoJoe replied:
yeah sure  I will change it this way.

ParsaYadollahi commented on dsc-website-client/src/components/CustomizedCard.js:
```diff
@@ -0,0 +1,129 @@
+          <ExpandMoreIcon />
+        </IconButton>
+      </CardActions>
+      <Collapse in={expanded} timeout="auto" unmountOnExit>
```
Do we need this expand feature?

ChangjunZhoJoe replied:
I am not sure how many words are there going to be in one Card? If there are two many we probably need the expand feature, otherwise the card with many words is going to take lots of space.

ParsaYadollahi replied:
But for each events wouldn't we just have a brief description of the event? I don't see how there would be a ton to talk about the event. Let me know.

ParsaYadollahi commented on dsc-website-client/src/components/CustomizedCard.js:
```diff
@@ -0,0 +1,129 @@
+      </CardContent>
+      <CardActions disableSpacing>
+        <IconButton aria-label="add to favorites">
+          <FavoriteIcon />
```
Do we need this like and share button? What purpose would they hold?

ChangjunZhoJoe replied:
That's true we don't actually need those buttons. Will take them down

ParsaYadollahi reviewed (changes requested):
I like the fact that it's almost all react and material ui, not much html. The code is clean but there are small fixes here and there that could be made. Overall, nice work!

ChangjunZhoJoe committed:
modified version

ChangjunZhoJoe committed:
alpha version of the timeline

ChangjunZhoJoe committed:
Merge branch 'master' into TimeLinePage

ChangjunZhoJoe committed:
refactor

ChangjunZhoJoe committed:
"merge conflicts"

Merge branch 'TimeLinePage' of https://github.com/dscmcgill/website into TimeLinePage

ChangjunZhoJoe committed:
refactor
